packages/components/src/components/Feedback/Feedback.tsx

```diff
@@ -0,0 +1,180 @@
+import { Icon } from '../Icon';
+import { Label, Paragraph } from '../Typography';
+import styled, { css } from 'styled-components';
+import { Spinner } from '../Spinner';
+import {
+  Thumbdown,
+  ThumbdownFilled,
+  Thumbup,
+  ThumbupFilled,
+} from '../../icons/tsx';
+import { TextArea } from '../TextArea';
+import { ddsBaseTokens } from '@norges-domstoler/dds-design-tokens';
+
+type RatingContainerProps = {
+  layout: Layout;
+};
+
+type FlexContainerProps = {
+  flexDirection: string;
+};
+
+const RatingContainer = styled.div<RatingContainerProps>`
+  display: flex;
+  gap: ${ddsBaseTokens.spacing.SizesDdsSpacingLocalX05};
+  ${({ layout }) => css`
+    flex-direction: ${layout === 'horizontal' ? 'row' : 'column'};
+    align-items: ${layout === 'horizontal' ? 'center' : 'start'};
+  `}
+`;
+
+const FlexContainer = styled.div<FlexContainerProps>`
+  display: flex;
+  ${({ flexDirection }) => css`
+    flex-direction: ${flexDirection};
+    gap: ${flexDirection === 'row'
+      ? ddsBaseTokens.spacing.SizesDdsSpacingLocalX05
+      : ddsBaseTokens.spacing.SizesDdsSpacingLocalX1};
+  `};
+`;
+
+const IconLabelSpan = styled.span`
+  display: inline-flex;
+  align-items: center;
+  gap: ${ddsBaseTokens.spacing.SizesDdsSpacingLocalX05};
+`;
+
+export const Feedback = ({
+  layout = 'vertical',
+  ratingLabel,
+  positiveFeedbackLabel = 'Hva kan vi forbedre? (valgfritt)',
+  negativeFeedbackLabel = 'Hva kan vi forbedre? (valgfritt)',
+  ratingValue,
+  feedbackTextValue,
+  thumbUpTooltip = 'Bra',
+  thumbDownTooltip = 'Dårlig',
+  feedbackTextAreaExcluded = false,
+  loading = false,
+  isSubmitted,
+  onRating,
+  onFeedbackTextChange,
+  onSubmit,
+}: FeedbackProps) => {
+  const [rating, setRating] = useState<Rating | null>(null);
+  const [feedbackText, setFeedbackText] = useState<string>();
+  const [isFeedbackSubmitted, setIsFeedbackSubmitted] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    ratingValue !== undefined && setRating(ratingValue);
+  }, [ratingValue, setRating]);
+
+  useEffect(() => {
+    feedbackTextValue !== undefined && setFeedbackText(feedbackTextValue);
+  }, [feedbackTextValue, setFeedbackText]);
+
+  useEffect(() => {
+    isSubmitted !== undefined && setIsFeedbackSubmitted(isSubmitted);
+  }, [isSubmitted, setFeedbackText]);
+
+  const handleRatingChange = (newRating: Rating) => {
+    onRating && onRating(newRating);
+    onSubmit && feedbackTextAreaExcluded && onSubmit(newRating, '');
+
+    ratingValue === undefined && setRating(newRating);
+  };
+
+  const handleFeedbackTextChange = (newFeedbackText: string) => {
+    onFeedbackTextChange && onFeedbackTextChange(newFeedbackText);
+    feedbackTextValue === undefined && setFeedbackText(newFeedbackText);
+  };
+
+  const handleSubmit = () => {
+    onSubmit && onSubmit(rating!, feedbackText ?? '');
+    isSubmitted === undefined && setIsFeedbackSubmitted(true);
+  };
+
+  return (
+    <>
+      {rating === null && !isFeedbackSubmitted ? (
```

Comment:
Disse toppnivå-terneries burde flyttes ut til å bare være vanlig if-statements:

```tsx
if(rating === null && !isFeedbackSubmitted) {
  return (
    <RatingContainer layout={layout}>
      ...
  )
}

if(!feedbackTextAreaExcluded && !isFeedbackSubmitted) {
  return (
    <FlexContainer flexDirection="column">
    ...
  )
}

return (
  <Paragraph>Takk for tilbakemeldingen!</Paragraph>
)
```

Det blir litt enklere å lese.

Reply:
Hver sti (bortsett fra "Takk for tilbakemelding" siden den er så enkel) burde nok også være egne komponenter siden de er såpass store. Synes det blir vanskelig å lese når det er så mye jsx i en komponent 😅 Vil bare presisere at de ikke skal eksporteres for eksterne konsumenter da 😄